### PR TITLE
[SYCL-MLIR] Add `polygeist.struct` type

### PR DESCRIPTION
**Pull request by whitneywhtsang**
We want to have a structure type that is allowed to be the element type of MemRef.

### CODE REVIEW

File: polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistTypes.td [hunk 1]
```diff
@@ -25,4 +25,15 @@ class Polygeist_Type<string name, string typeMnemonic, list<Trait> traits = []>
   let mnemonic = typeMnemonic;
 }
 
+def Polygeist_StructType :
+    Polygeist_Type<"Struct", "struct"> {
+  let summary = "Polygeist Struct";
+  let description = [{
+    Structure type that allowed to be the element type of MemRef and LLVM
```
Comment (etiotto) [inline polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistTypes.td:32]: [typo]: Structure type that *is* allowed ...

**Review by etiotto — APPROVED** [review]

File: polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistTypes.td (hunk 1 above)
Comment (whitneywhtsang) [reply polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistTypes.td]: done.

**Review by sommerlukas — APPROVED** [review]

**Comment by sommerlukas** [comment]
> Not sure what the rest think, @sommerlukas @etiotto, but I think an optional `typeName` would be nice to have

Yes, that would indeed improve readability. 

Also, the ability to define aliases and have less literal struct types inside the code would definitely improve readibility. But that of course does not need to be part of this PR.

**Comment by sommerlukas** [comment]
> Not sure what the rest think, @sommerlukas @etiotto, but I think an optional `typeName` would be nice to have

Yes, that would indeed improve readability. 

Also, the ability to define aliases and have less literal struct types inside the code would definitely improve readibility. But that of course does not need to be part of this PR.

**Comment by whitneywhtsang** [comment]
> > Not sure what the rest think, @sommerlukas @etiotto, but I think an optional `typeName` would be nice to have
> 
> Yes, that would indeed improve readability.
> 
> Also, the ability to define aliases and have less literal struct types inside the code would definitely improve readibility. But that of course does not need to be part of this PR.

Added optional name.

**Review by victor-eds — APPROVED** [review]

**Pull request merged**